File: nodes/random-error/random-error.process.py
import random

async def process(inputs, settings, config, nodeConfig):
    """
    Process function for the Random Error node.
    Randomly raises an exception based on probability, or passes through the value.
    """
    value = inputs.get("value")
    error_probability = float(inputs.get("errorProbability", 0.1))  # Default to 10% chance
    error_message = inputs.get("errorMessage", "Random error occurred")
    error_type = inputs.get("errorType", "RandomError")
    
    # Validate probability is between 0 and 1
    probability = max(0.0, min(1.0, error_probability))
    
    # Generate random number between 0 and 1
    random_val = random.random()
    
    # Raise error if random number is less than probability
    if random_val < probability:
        class RandomError(Exception):
            def __init__(self, message, probability, random_value):
                super().__init__(message)
                self.name = error_type
                self.probability = probability
                self.random_value = random_value
        
        raise RandomError(error_message, probability, random_val)
    
    # Otherwise, pass through the value
    return {"result": value} 

File: sdks/nodejs/nodes/random-error/random-error.process.js
export default async ({inputs, settings, config}) => {

  const value = inputs.value;
  const errorProbability = parseFloat(inputs.errorProbability) || 0.1; // Default to 10% chance
  const errorMessage = inputs.errorMessage || "Random error occurred";
  const errorType = inputs.errorType || "RandomError";
  
  // Validate probability is between 0 and 1
  const probability = Math.max(0, Math.min(1, errorProbability));
  
  // Generate random number between 0 and 1
  const random = Math.random();
  
  // Throw error if random number is less than probability
  if (random < probability) {
    const error = new Error(errorMessage);
    error.name = errorType;
    error.probability = probability;
    error.randomValue = random;
    throw error;
  }
  
  // Otherwise, pass through the value
  return { result: value };
}; 